components/channel_header/components/header_icon_wrapper.tsx

```diff
@@ -22,7 +22,7 @@ type Props = {
     iconComponent: React.ReactNode;
     onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
     tooltipKey: string;
-    tooltipText?: React.ReactNode;
+    tooltipText?: string;
```

Comment:
I would say keep both because I think there was a reasoning behind that type. 

```suggestion
    tooltipText?: React.ReactNode | string;
```

Reply:
I have seen this before for being able to send over the `FormattedText` component. In this particular case, we sadly accept according to the documentation any react component can be passed here :(

https://developers.mattermost.com/integrate/plugins/webapp/reference/#registerChannelHeaderButtonAction

I would deprecate that, things I don't think any plugin is using a component there, but we cannot be sure.

Reply:
A possible solution is to check whether it is a string or not, and if it is not a string, use the plugin id as a fallback.

Reply:
Thanks @larkox for referencing the docs, i see the point now. @nevyangelova `ReactNode` combination of many types including `String`.
I updated as per suggestion.